sdk/scheduler/src/main/java/org/apache/mesos/offer/constrain/MaxPerAttributeRule.java

```diff
@@ -48,22 +48,56 @@
  * enforce the limit value of 5. In this case, we see that rack:a is full but that rack:b and rack:c
  * still have room. Therefore the PlacementRule will allow deployments given an attribute of
  * 'rack:b' or 'rack:c', and will block deployments onto 'rack:a'.
+ *
+ * In addition, this enforcement can be selectively applied by task name. By default the rule will
+ * look at ALL tasks in the service, but with a task matcher we can only count e.g. tasks named
+ * 'index-.*'. This allows us to only enforce the rule against certain task types or task instances
+ * within the service.
  */
 public class MaxPerAttributeRule implements PlacementRule {
 
     private final int maxTasksPerSelectedAttribute;
-    private final AttributeSelector attributeSelector;
+    private final StringMatcher attributeMatcher;
+    private final StringMatcher taskFilter;
```

Comment:
The tricky part is that taskFilter is being applied to the current state of known tasks (3rd parameter), while `*Rule` operate against the Offer being evaluated (1st parameter). They're each effectively performing work against different sources of information.